odra-macros/src/ir/ref_item.rs

```diff
@@ -0,0 +1,152 @@
+use super::checked_unwrap;
+use crate::ir::{FnIR, ModuleIR};
+use proc_macro2::TokenStream;
+use quote::{quote, ToTokens};
+use syn::parse_quote;
+
+pub struct RefItem<'a> {
+    module: &'a ModuleIR
+}
+
+impl<'a> RefItem<'a> {
+    pub fn new(module: &'a ModuleIR) -> Self {
+        RefItem { module }
+    }
+
+    pub fn method(&self, fun: &FnIR) -> syn::ItemFn {
+        let fun_name = fun.name();
+        let fun_name_str = fun.name_str();
+        let args = fun.arg_names();
+        let typed_args = {
+            let args = fun.typed_args();
+            quote!(#(, #args)*)
+        };
+
+        let args = if fun.args_len() == 0 {
+            quote!(odra2::types::RuntimeArgs::new())
+        } else {
+            let args = args
+                .iter()
+                .map(|i| quote!(let _ = named_args.insert(stringify!(#i), #i);))
+                .collect::<TokenStream>();
+            quote!({
+                let mut named_args = odra2::types::RuntimeArgs::new();
+                #args
+                named_args
+            })
+        };
+
+        let return_type = fun.return_type();
+        let mutability = fun.is_mut().then(|| quote!(mut));
+        parse_quote!(
+            pub fn #fun_name(& #mutability self #typed_args) #return_type {
+                self.env.call_contract(
+                    self.address,
+                    CallDef::new(String::from(#fun_name_str), #args),
```

Comment:
You should use full paths to `CallDef` and `String` (from prelude).

Reply:
Yes, I discovered it in the meantime, It's already fixed, but will be added  to the next pr.